src/column.rs

```diff
@@ -143,9 +145,20 @@ impl Column {
 		})
 	}
 
+	// [Review] would it make sense to allow different hash than blake2b.
+	// -> most likely, we only use the 64 first bit and don't really need crypto
+	// resistance, or am I missing something.
+	// Actually this would be true if we did store the full key (but we store the
+	// hash).
+	// Note storing full key should probably be done by writing it into value
+	// (then there is the issue of having weak hash and a collision: requires
+	// to check the stored value or have an api that returns multiple matches).
```

Comment:
There's an option to store raw keys per column and avoid double hashing. But in generals, having fixed size keys made for simple design. Hashing allows the user to disregard any potential collisions, although by itself it is not enough to protect against a malicious attacker

In general, I'd prefer no to add features, such as alternative hashing functions, or arbitrary key sizes, unless they are really required by substrate. The focus here is to  maximise state database efficiency above all else, while keeping the rest reasonably fast. State is by far our largest data set.